webapp/components/wpt-metadata.js

```diff
@@ -0,0 +1,198 @@
+import {
+  html,
+  PolymerElement
+} from '../node_modules/@polymer/polymer/polymer-element.js';
+import { LoadingState } from './loading-state.js';
+
+class WPTMetadataNode extends PolymerElement {
+  static get template() {
+    return html`
+      <style>
+        .metadataNode {
+          display: flex;
+          align-items: center;
+          margin-bottom: 4px;
+        }
+        .metadataNode img {
+          margin-right: 16px;
+          height: 24px;
+          width: 24px;
+        }
+      </style>
+      <div class="metadataNode">
+        <iron-icon icon="bug-report"></iron-icon>
+        <div>
+          [[metadataNode.test]] :
+          <a href="[[href]]">[[metadataNode.url]]</a>
+          <br />
+        </div>
+      </div>
+    `;
+  }
+
+  static get is() {
+    return 'wpt-metadata-node';
+  }
+
+  static get properties() {
+    return {
+      metadataNode: Object,
+      href: {
+        type: String,
+        computed: 'computeHref(metadataNode)'
+      }
+    };
+  }
+
+  computeHref(metadataNode) {
+    const prefix = 'https://';
+    if (metadataNode.url.substr(0, prefix.length) !== prefix) {
+      return prefix + metadataNode.url;
+    }
+    return metadataNode.url;
+  }
+}
+window.customElements.define(WPTMetadataNode.is, WPTMetadataNode);
+
+class WPTMetadata extends LoadingState(PolymerElement) {
+  static get template() {
+    return html`
+      <style>
+        h4 {
+          margin-bottom: 0.5em;
+        }
+      </style>
+      <template is="dom-if" if="[[firstThree]]">
+        <h4>Triaged Metadata in <i>[[path]]</i></h4>
+      </template>
+      <template is="dom-repeat" items="[[firstThree]]" as="metadataNode">
+        <wpt-metadata-node metadata-node="[[metadataNode]]"></wpt-metadata-node>
+      </template>
+      <template is="dom-if" if="[[others]]">
+        <iron-collapse id="metadata-collapsible">
+          <template is="dom-repeat" items="[[others]]" as="metadataNode">
+            <wpt-metadata-node
+              metadata-node="[[metadataNode]]"
+            ></wpt-metadata-node>
+          </template>
+        </iron-collapse>
+        <paper-button id="metadata-toggle" onclick="[[openCollapsible]]">
+          Show more
+        </paper-button>
+      </template>
+      <br>
+    `;
+  }
+
+  static get is() {
+    return 'wpt-metadata';
+  }
+
+  static get properties() {
+    return {
+      products: {
+        type: Array,
+        observer: 'loadAllMetadata'
+      },
+      path: String,
+      metadata: Array,
+      displayedMetadata: {
+        type: Array,
+        computed: 'computeDisplayedMetadata(path, metadata)'
+      },
+      firstThree: {
+        type: Array,
+        computed: 'computeFirstThree(displayedMetadata)'
+      },
+      others: {
+        type: Array,
+        computed: 'computeOthers(displayedMetadata)'
+      }
+    };
+  }
+
+  constructor() {
+    super();
+    this.openCollapsible = this.handleOpenCollapsible.bind(this);
```

Comment:
Do we need to declare `openCollapsible` as a property (with type function)?

Reply:
openCollapsible is used strictly within this wpt-metadata custom element. I think as long as openCollapsible is not invoked and changed outside of this class, it doesn't have to be declared as a property